app/models/vm_or_template.rb

```diff
@@ -1833,6 +1837,21 @@ def parent_resource
     parent
   end
 
+  def available_child_options
+    Rbac.filtered(VmOrTemplate.by_ems(ems_id)
+      .where.not(:id => id)
+      .where.not(:id => children.pluck(:ids))
+      .where.not(:id => parent.id))
```

Comment:
you can do
`VmOrTemplate.by_ems(ems_id) .where.not(:id => [id] + children.pluck(:ids) )`


`parent_id` :

(1) it sounds reasonable and It will be probably better to use

`.where.not(:id => parents))` (when some VM has multiple parents method `parent` will throw error )  

and (2) I see similar issue in `available_parent_options` - here should not be children( I guess) in selection of parent. 

but what about to do (1) and (2)  in follow up and keep this just PR only for  moving code from UI to core ? 

thanks!

Reply:
> (when some VM has multiple parents method `parent` will throw error )

LOL @Fryguy Did we just find the source of the "multiple parents found" issue that we've never been able to reproduce?

Reply:
@bdunne that would be so nice if we solved that one once and for all.

maybe @Ladas can look at the assignment logic and see if it matches the logic he used in refresh. I do remember that depending upon how you did your assignments, it could cause problems. Something like assigning via children or parents is better. But my memory is vague.

Reply:
Hm this logic was custom anyway, since the assignment must be done per 1 record, it's very ineffective. (means 1 query per 1 record). I believe we assign parent to children now, since we save it from the children view.